packages/opentelemetry-sampler-aws-xray/test/remote-sampler.test.ts

```diff
@@ -0,0 +1,174 @@
+        SamplingRule: {
+          Attributes: {},
+          FixedRate: 0.05,
+          HTTPMethod: '*',
+          Host: '*',
+          Priority: 10000,
+          ReservoirSize: 1,
+          ResourceARN: '*',
+          RuleARN: 'arn:aws:xray:us-west-2:123456789000:sampling-rule/Default',
+          RuleName: 'Default',
+          ServiceName: '*',
+          ServiceType: '*',
+          URLPath: '*',
+          Version: 1,
+        },
+      },
+      {
+        CreatedAt: 1.676038494e9,
+        ModifiedAt: 1.676038494e9,
+        SamplingRule: {
+          Attributes: {},
+          FixedRate: 0.2,
+          HTTPMethod: 'GET',
+          Host: '*',
+          Priority: 1,
+          ReservoirSize: 10,
+          ResourceARN: '*',
+          RuleARN: 'arn:aws:xray:us-west-2:123456789000:sampling-rule/Rule2',
+          RuleName: 'Rule2',
+          ServiceName: 'FooBar',
+          ServiceType: '*',
+          URLPath: '/foo/bar',
+          Version: 1,
+        },
+      },
+    ],
+  };
+
+  let clock: sinon.SinonFakeTimers;
+  let sampler: AWSXRayRemoteSampler;
+  let axiosPostSpy: sinon.SinonSpy;
+
+  const defaultEndpoint = 'http://localhost:2000';
+  const pollingInterval = 60 * 1000;
+  const config = {
+    endpoint: defaultEndpoint,
+    pollingInterval: pollingInterval,
+  };
+
+  before(() => {
+    nock('http://localhost:2000')
+      .persist()
+      .post('/GetSamplingRules')
+      .reply(200, getSamplingRulesResponseStub);
+  });
+
+  beforeEach(() => {
+    clock = sinon.useFakeTimers();
+    axiosPostSpy = sinon.spy(axios, 'post');
+    sampler = new AWSXRayRemoteSampler(config);
+  });
+
+  afterEach(() => {
+    clock.restore();
+    axiosPostSpy.restore();
+  });
+
+  it('should throw TypeError when an invalid polling interval is passed in', async () => {
+    const configWithZeroPollingInterval = {
+      endpoint: 'http://localhost:2000',
+      pollingInterval: 0,
+    };
+    const configWithNegativeInterval = {
+      endpoint: 'http://localhost:2000',
+      pollingInterval: -5,
+    };
+
+    assert.throws(
+      () => new AWSXRayRemoteSampler(configWithZeroPollingInterval),
+      TypeError
+    );
+    assert.throws(
+      () => new AWSXRayRemoteSampler(configWithNegativeInterval),
+      TypeError
+    );
+  });
+
+  it('should make a POST request to the /GetSamplingRules endpoint', async () => {
+    clock.tick(pollingInterval);
+    sinon.assert.calledOnce(axiosPostSpy);
+  });
+
+  it('should make 3 POST requests to the /GetSamplingRules endpoint after 3 intervals have passed', async () => {
+    clock.tick(pollingInterval);
+    clock.tick(pollingInterval);
+    clock.tick(pollingInterval);
+
+    sinon.assert.calledThrice(axiosPostSpy);
+  });
+
+  it('should initialize endpoint and polling interval from config correctly', async () => {
+    assert.strictEqual(
+      sampler.toString(),
+      `AWSXRayRemoteSampler{endpoint=${
+        defaultEndpoint + '/GetSamplingRules'
```

Comment:
Since the value you use in your configuration is identical to the one being set as default by the component, this test doesn't actually check what it claims. If the component is ignoring the config and always takes the default, your test will still pass.

I think `defaultEndpoint` should be set to something else than `DEFAULT_ENDPOINT`, like `http://localhost:1234` to assert it is indeed used